src/server/repository/assessmentCycle/country/getCountryLastPublishedInfo.ts

```diff
@@ -17,24 +17,30 @@ export const getCountryLastPublishedInfo = async (
   const selectStatements = assessment.cycles
     .map(
       (cycle) =>
-        `select country_iso, last_in_published, '${cycle.uuid}' as cycle_uuid, '${cycle.name}' as cycle_name
+        `select country_iso, 
+         last_in_published,
+         last_update,
+         '${cycle.uuid}' as cycle_uuid, 
+         '${cycle.name}' as cycle_name
          from ${Schemas.getNameCycle(assessment, cycle)}.country ${countryIso ? 'where country_iso = $1' : ''}`
     )
     .join(' union all ')
 
+  // Note: lastPublished: from last published country (order by ...last_in_published....)
+  // we find the greatest of last update and published timestamp
   const query = `
     with rows as (${selectStatements})
     select jsonb_object_agg(
       country_iso,
       jsonb_build_object(
               'cycleUuid', cycle_uuid,
               'cycleName', cycle_name,
-              'lastPublished', last_in_published
+              'lastPublished', greatest(last_in_published, last_update)
       )
     ) as result
     from (
       select distinct on (country_iso)
-        country_iso, cycle_uuid, cycle_name, last_in_published
+        country_iso, cycle_uuid, cycle_name, last_in_published, last_update
       from rows
       order by country_iso, last_in_published desc nulls last
```

Comment:
@minotogna Now I updated and tested:

When a published country (test case, FIN published) gets edited by admin, the timestamp is correctly updated
To fetch the correct country, we look for the latest _published_ country

It should be ok now

Let me know